client/lib/logmein/index.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Append logmein=1 query parameter to mapped domain urls we want the user to be logged in against.
+ */
+
+// Used as placeholder / default domain to detect when we're looking at a relative url
+const INVALID_URL = `http://__domain__.invalid`;
+
+export function logmeinUrl( fullUrl: string, allow: string[] ): string {
+	let url: URL;
+
+	try {
+		url = new URL( String( fullUrl ), INVALID_URL );
+	} catch ( e ) {
+		// Ignore unparseable urls
+		return fullUrl;
+	}
+
+	// Ignore and passthrough /relative/urls that have no host specified
+	if ( url.origin === INVALID_URL ) {
+		return fullUrl;
+	}
+
+	// Ignore urls not in the allow list
+	if ( allow.indexOf( url.origin ) === -1 ) {
+		return fullUrl;
+	}
+
+	url.searchParams.set( 'logmein', 'direct' );
+	return url.toString();
+
+	// Todo: I wasn't able to make either of these work, thoughts?
+	// return (
+	// 	'https://wordpress.com/log-in?redirect_to=' +
+	// 	encodeURIComponent(
+	// 		'https://r-login.wordpress.com/remote-login.php?action=link&back=' +
+	// 			encodeURIComponent( fullUrl )
+	// 	)
+	// );
+
+	// return (
+	// 	'https://r-login.wordpress.com/remote-login.php?action=link&back=' +
+	// 	encodeURIComponent( fullUrl )
```

Comment:
This should work, since I have a similar implementation in https://github.com/Automattic/wp-calypso/pull/53125/ and it works well. One thing to note is that the back URL should be `https` instead of `http`. When I tested this PR by clicking on the mapped domain link on the domain management page, I was taken to the following URL which [outputted JS on the screen](https://d.pr/i/KbqF2a):

```
https://r-login.wordpress.com/remote-login.php?action=link&back=http%3A%2F%2Fniranjan792177516.blog
```



Note the `http` in the back URL. When I changed that `https`, then the r-login redirect worked well. 

Anyhow, with the change in D61461-code to skip the Continue step, is this change still needed?